src/propTypes.js

```diff
@@ -102,11 +103,14 @@ const createShapeTypeChecker = type => shapeObj => {
 };
 
 const componentTypeChecker = name => {
-    const appliedChecker = PropTypes.shape({ name });
+    const appliedChecker = PropTypes.shape();
+    let allowedComponents;
+    if(name) allowedComponents = [ name ];
```

Comment:
```suggestion
    if (name) {
        allowedComponents = [ name ];
    }    
```
maybe just no lint on this project yet